services/director-v2/tests/unit/test_modules_dynamic_sidecar_observer.py

```diff
@@ -0,0 +1,161 @@
+# pylint:disable=protected-access
+# pylint:disable=redefined-outer-name
+# pylint:disable=unused-argument
+
+from typing import Optional
+from unittest.mock import AsyncMock
+
+import pytest
+from fastapi import FastAPI
+from pytest import MonkeyPatch
+from pytest_mock.plugin import MockerFixture
+from pytest_simcore.helpers.typing_env import EnvVarsDict
+from simcore_service_director_v2.core.settings import AppSettings
+from simcore_service_director_v2.models.schemas.dynamic_services import SchedulerData
+from simcore_service_director_v2.modules.dynamic_sidecar.api_client import (
+    setup,
+    shutdown,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.scheduler import (
+    DynamicSidecarsScheduler,
+    setup_scheduler,
+    shutdown_scheduler,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core._observer import (
+    _apply_observation_cycle,
+)
+
+
+@pytest.fixture
+def disable_observation(mocker: MockerFixture) -> None:
+    mocker.patch(
+        "simcore_service_director_v2.modules.dynamic_sidecar.scheduler._task.DynamicSidecarsScheduler.start",
+        autospec=True,
+    )
+
+
+@pytest.fixture
+def mock_are_sidecar_and_proxy_services_present(mocker: MockerFixture) -> None:
+    mocker.patch(
+        "simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core._observer.are_sidecar_and_proxy_services_present",
+        autospec=True,
+        return_value=False,
+    )
+
+
+@pytest.fixture
+def mock_events(mocker: MockerFixture) -> None:
+    for event_to_mock in (
+        "CreateSidecars",
+        "WaitForSidecarAPI",
+        "UpdateHealth",
+        "GetStatus",
+        "PrepareServicesEnvironment",
+        "CreateUserServices",
+        "AttachProjectsNetworks",
+        "RemoveUserCreatedServices",
+    ):
+        mocker.patch(
+            f"simcore_service_director_v2.modules.dynamic_sidecar.scheduler._core._events.{event_to_mock}.action",
+            autospec=True,
+            return_value=True,
+        )
+
+
+@pytest.fixture
+def mock_env(
+    docker_swarm: None,
+    mock_env: EnvVarsDict,
+    monkeypatch: MonkeyPatch,
+) -> None:
+    monkeypatch.setenv("SIMCORE_SERVICES_NETWORK_NAME", "test_network")
```

Comment:
FYI: you can use ``setenvs_from_dict`` here

https://github.com/ITISFoundation/osparc-simcore/blob/b88565c1814061ff1e2d8cfdcf601fb1c2f33350/packages/pytest-simcore/src/pytest_simcore/helpers/utils_envs.py#L21